server/api.go

```diff
@@ -25,17 +25,31 @@ import (
 
 var Logger *logrus.Logger = logrus.StandardLogger()
 
+const LDContextSessionInfo = "https://irma.app/ld/request/info/v1"
+const LDContextSessionOptions = "https://irma.app/ld/options/v1"
+
 type SessionPackage struct {
-	SessionPtr *irma.Qr `json:"sessionPtr"`
-	Token      string   `json:"token"`
+	SessionPtr    *irma.Qr           `json:"sessionPtr"`
+	Token         irma.BackendToken  `json:"token"`
+	FrontendToken irma.FrontendToken `json:"frontendToken"`
+}
+
+// SessionInfo contains all information irmaclient needs to know to initiate a session.
+// Session request is stored as RawMessage since it is not used internally and otherwise we would
+// have to make a manual JSON marshal function with type checking since SessionRequest is an interface.
+type SessionInfo struct {
```

Comment:
Ah I thought the division was that messages initiated from external sources where in `messages.go` and all messages initiated by the server where in the server's `api.go`. For example, the `SessionResult` and `Status` where also in there. Just coincidence then that that interpretation was also possible.

It is not that much of a work to change it, with ctrl-f we can already find a lot. I thought there are no urls derived from this name anymore, so that should be all right. The name also bothered me a bit, but I didn't know anything better. `ClientRequest`  is maybe better indeed.

With moving it to `messages.go` we kind of create an inconsistance, namely that `Status` now still remains part of `server` and when we use your interpretation of separation then also this type should be part of `messages.go`. `messages.go` however already contains a `Status` type for the client side perspective. Since the string values of both Status struct are different (capitalized and not-capitalized), we cannot simply merge both. There are also some statuses only relevant for clients there. Then we have to rename the [status names as communicated to the client](https://github.com/privacybydesign/irmago/blob/master/messages.go#L168) too maybe. How do you look to this?

Reply:
Okay. I moved `ClientRequest` to the `irma` package now. This was indeed beter, because otherwise the `JSONUnmarshall` function would be in the `server` package while it is only needed in the `irmaclient` package. This would be a very strange situation. 

Only question left is what to do with the `Status` type. Can we somehow merge those two types into one general status types? Maybe with a legacy version for older protocol versions?

Reply:
(Note to self): If the status eventually moves to the `irma` package, also the `WaitStatus` functions must move.

Reply:
Yes, I think the `Status` should also move. I don't think we should merge the statuses into one type. We can just rename `irma.Status` to `irma.ClientStatus`, and then call this one `irma.ServerStatus` (without touching the string values of any of these).